src/EnergyPlus/HVACUnitaryBypassVAV.cc

```diff
@@ -2244,84 +2126,58 @@ namespace HVACUnitaryBypassVAV {
-        int MixerOutsideAirNode; // Outside air node number in OA mixer
-        int MixerReliefAirNode;  // Relief air node number in OA mixer
-        int DehumidMode;         // Dehumidification mode (0=normal, 1=enhanced)
+        Real64 MinHumRat;       // Minimum humidity ratio for sensible capacity calculation (kg/kg)
+        Array1D<Real64> Par(6); // RegulaFalsi parameters
+        int SolFla;             // Flag of RegulaFalsi solver
+        Real64 QHeater;         // Load to be met by heater [W]
+        Real64 QHeaterActual;   // actual heating load met [W]
+        Real64 CpAir;           // Specific heat of air [J/kg-K]
+        int DehumidMode;        // Dehumidification mode (0=normal, 1=enhanced)
         Real64 ApproachTemp;
         Real64 DesiredDewPoint;
         Real64 OutdoorDryBulbTemp; // Dry-bulb temperature at outdoor condenser
         Real64 OutdoorBaroPress;   // Barometric pressure at outdoor condenser
         // FLOW
 
-        OutletNode = CBVAV(CBVAVNum).AirOutNode;
-        InletNode = CBVAV(CBVAVNum).AirInNode;
-        MixerOutsideAirNode = CBVAV(CBVAVNum).MixerOutsideAirNode;
-        MixerReliefAirNode = CBVAV(CBVAVNum).MixerReliefAirNode;
+        int OutletNode = CBVAV(CBVAVNum).AirOutNode;
+        int InletNode = CBVAV(CBVAVNum).AirInNode;
         if (CBVAV(CBVAVNum).CondenserNodeNum > 0) {
-            OutdoorDryBulbTemp = Node(CBVAV(CBVAVNum).CondenserNodeNum).Temp;
-            OutdoorBaroPress = Node(CBVAV(CBVAVNum).CondenserNodeNum).Press;
+            OutdoorDryBulbTemp = DataLoopNode::Node(CBVAV(CBVAVNum).CondenserNodeNum).Temp;
+            OutdoorBaroPress = DataLoopNode::Node(CBVAV(CBVAVNum).CondenserNodeNum).Press;
         } else {
-            OutdoorDryBulbTemp = OutDryBulbTemp;
-            OutdoorBaroPress = OutBaroPress;
+            OutdoorDryBulbTemp = DataEnvironment::OutDryBulbTemp;
+            OutdoorBaroPress = DataEnvironment::OutBaroPress;
         }
 
         SaveCompressorPLR = 0.0;
 
         // Bypass excess system air through bypass duct and calculate new mixed air conditions at OA mixer inlet node
-        Node(CBVAV(CBVAVNum).MixerInletAirNode).Temp =
-            (1.0 - BypassDuctFlowFraction) * Node(InletNode).Temp + BypassDuctFlowFraction * Node(OutletNode).Temp;
-        Node(CBVAV(CBVAVNum).MixerInletAirNode).HumRat =
-            (1.0 - BypassDuctFlowFraction) * Node(InletNode).HumRat + BypassDuctFlowFraction * Node(OutletNode).HumRat;
-        Node(CBVAV(CBVAVNum).MixerInletAirNode).Enthalpy =
-            PsyHFnTdbW(Node(CBVAV(CBVAVNum).MixerInletAirNode).Temp, Node(CBVAV(CBVAVNum).MixerInletAirNode).HumRat);
-        SimOAMixer(CBVAV(CBVAVNum).OAMixName, FirstHVACIteration, CBVAV(CBVAVNum).OAMixIndex);
-
-        if (CBVAV(CBVAVNum).FanPlace == BlowThru) {
+        if (CBVAV(CBVAVNum).plenumIndex > 0 || CBVAV(CBVAVNum).mixerIndex > 0) {
+            Real64 saveMixerInletAirNodeFlow = DataLoopNode::Node(CBVAV(CBVAVNum).MixerInletAirNode).MassFlowRate;
```

Comment:
Must save operating flow rate at mixer inlet node since the system inlet node will sum to the sum of the TU's plus the bypass. The first pass at FirstHVACIteration = true sets full flow and bypass = 0 so the sum of the TU's will be less than full flow (i.e., they won't be at full flow once they simulate). If lines 2167 and 2169 are removed, there are warnings of low flow per capacity warnings for DX coil, indicating that the flow through the coil is lower than expected, hence operating flow needs to be saved. Lines 2168 and 2169, pass the node conditions from inlet to the mixer inlet and reset mass flow.